Add support for multi-root workspaces

vianney opened the pull request:
This patch adds multi-root workspaces saved as .code-workspace files to the results list.

Note: this is my first foray into rust-land, so any comment is more than welcome.

Closes #14

lunaryorn commented on src/main.rs:
```diff
@@ -27,12 +27,20 @@ use gnome_search_provider_common::log::*;
     folder_uri: Option<String>,
     #[serde(rename = "fileUri")]
     file_uri: Option<String>,
+    workspace: Option<StorageOpenedPathsListEntryWorkspace>,
```
I presume it's either a folder, or a file, or a workspace?  So perhaps you could turn this into an untagged enum? See https://serde.rs/enum-representations.html#untagged

lunaryorn commented on src/main.rs:
```diff
@@ -27,12 +27,20 @@ use gnome_search_provider_common::log::*;
 use gnome_search_provider_common::matching::*;
 
+#[derive(Debug, Deserialize)]
+struct StorageOpenedPathsListEntryWorkspace {
```
I think `WorkspaceEntry` would do :slightly_smiling_face:

lunaryorn commented on src/main.rs:
```diff
@@ -27,12 +27,20 @@ use gnome_search_provider_common::log::*;
+struct StorageOpenedPathsListEntryWorkspace {
+    id: Option<String>,
+    #[serde(rename = "configPath")]
+    config_path: Option<String>,
```
Why's this optional?  How would a workspace _without_ configuration path even work?  Does this really exist in VSCode?

lunaryorn commented on src/main.rs:
```diff
@@ -74,7 +82,13 @@ impl Storage {
                 .into_iter()
-                .filter_map(|entry| entry.folder_uri)
+                .filter_map(|entry| {
+                    if let Some(workspace) = entry.workspace {
```
That's `entry.workspace.and_then(|w| w.config_path).or(entry.folder_uri)` :slightly_smiling_face:

lunaryorn reviewed (changes requested):
Thanks :pray:   I've left a few comments, and I'd like to ask you to add a `CHANGELOG.md` entry for this change and the corresponding issue.

vianney commented:
Thanks a lot for the review and letting me learn rust on the go. I have updated my branch with the requested changes, and updated the test suite as well.

lunaryorn commented:
@vianney Thanks a bunch :pray:
